faunadb-java-dsl/src/main/java/com/faunadb/client/types/Field.java

```diff
@@ -106,6 +155,32 @@ public CollectionCodec(Path path, Field<A> field) {
     return new Field<>(Path.empty(), codec);
   }
 
+  /**
+   * Creates a field that converts its underyling value to the {@link Class} provided.
+   *
+   * @param <T> the desired final type
+   * @return a new {@link Field} instance
+   */
+  public static <T> Field<T> as(Class<T> clazz) {
+    return root().to(clazz);
+  }
+
+  /**
+   * Assuming the {@link Value} instance is a collection, creates a field extractor that collects
+   * the {@link Field} provided for each element in the underlying collection.
+   *
+   * @param <A> the desired final type for each collection element
+   * @param field the {@link Field} to be extracted from each collection element
+   * @return a new {@link Field} instance
+   */
+  public static <A> Field<List<A>> asListOf(Field<A> field) {
+    return new Field<>(Path.empty(), new CollectionCodec<>(Path.empty(), field));
+  }
+
+  public static <A> Field<Map<String, A>> asMapOf(Field<A> field) {
+    return new Field<>(Path.empty(), Codec.OBJECT).to(new MapCodec<>(Path.empty(), field));
```

Comment:
if you just do `new Field<>(Path.empty(), new MapCodec<>(Path.empty(), field))` like in `asListOf()`